### PR TITLE
Remove PERF_SAMPLE_CALLCHAIN logic from TRACEPOINT events

### PR DESCRIPTION
**Pull request by palmtenor**
As discussed in #1601, since we had the stack-trace table early-on, and no tools still uses the old `PERF_SAMPLE_CALLCHAIN`, this PR removes the creation of `perf_reader` for `TRACEPOINT` events (kprobe / uprobe / Kernel Tracepoint), and parsing logic in `perf_reader`.

Also added some comments in `libbpf.c`

### CODE REVIEW

File: src/cc/libbpf.c [hunk 1]
```diff
@@ -762,47 +766,38 @@ static int bpf_attach_tracing_event(int progfd, const char *event_path,
...
       fprintf(stderr, "perf_event_open(%s/id): %s\n", event_path, strerror(errno));
       return -1;
     }
   }
-  perf_reader_set_fd(reader, pfd);
 
-  if (perf_reader_mmap(reader, attr.type, attr.sample_type) < 0)
-    return -1;
-
-  if (ioctl(pfd, PERF_EVENT_IOC_SET_BPF, progfd) < 0) {
```
Comment (palmtenor) [inline src/cc/libbpf.c]: Previously, if the `pfd` was created here (falling back to `debugfs`) and the`ioctl` fails, no would would clean them up. This commit fixes the issue and documented that caller should do the clean-up.

File: src/cc/libbpf.c [hunk 2]
```diff
@@ -823,15 +817,16 @@ void *bpf_attach_kprobe(int progfd, enum bpf_probe_attach_type attach_type,
...
+  // Otherwise it will read the event ID from the path in buf, create the
+  // Perf Event event using that ID, and updated value of pfd.
+  if (bpf_attach_tracing_event(progfd, buf, -1 /* PID */, &pfd) == 0)
+    return pfd;
 
 error:
-  perf_reader_free(reader);
-  return NULL;
-
+  bpf_close_perf_event_fd(pfd);
```
Comment (yonghong-song) [inline src/cc/libbpf.c]: Do you need to check whether pfd >= 0 here? Is it possible pfd still equals to initial value -1?
The same for below a few places.

File: src/lua/bcc/bpf.lua [hunk 3]
```diff
@@ -55,6 +56,12 @@ function Bpf.static.cleanup()
 
   detach_all("kprobes", Bpf.static.open_kprobes)
   detach_all("uprobes", Bpf.static.open_uprobes)
+
+  for key, perf_buffer in pairs(Bpf.static.perf_buffers) do
+    libbcc.perf_reader_free(perf_buffer)
+    Bpf.static.perf_buffers[key] = nil
+  end
```
Comment (yonghong-song) [inline src/lua/bcc/bpf.lua]: Not a lua expert. Maybe after all perf_reader_free, just clear Bpf.static.perf_buffers?

File: src/cc/libbpf.c (hunk 2 above)
Comment (palmtenor) [reply src/cc/libbpf.c]: `bpf_close_perf_event_fd` already checks the FD inside:)

File: src/lua/bcc/bpf.lua (hunk 3 above)
Comment (palmtenor) [reply src/lua/bcc/bpf.lua]: Yeah, me neither:( I basically mimicked existing code, it compiled and tests pass, but that's all I know......

**Pull request merged**